pkg/messaging/pair/server.go

```diff
@@ -0,0 +1,84 @@
+package pair
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.nanomsg.org/mangos/v3/protocol"
+	"go.nanomsg.org/mangos/v3/protocol/pair"
+	"nodemon/pkg/storing/nodes"
+)
+
+func StartPairMessagingServer(ctx context.Context, nanomsgURL string, ns *nodes.Storage) (protocol.Socket, error) {
+	if len(nanomsgURL) == 0 || len(strings.Fields(nanomsgURL)) > 1 {
+		log.Printf("Invalid nanomsg IPC URL for pair socket'%s'", nanomsgURL)
+		return nil, errors.New("invalid nanomsg IPC URL for pair socket")
+	}
+	socketPair, err := pair.NewSocket()
+	if err != nil {
+		log.Printf("Failed to get new pair socket: %v", err)
+		return nil, err
+	}
+
+	if err := socketPair.Listen(nanomsgURL); err != nil {
+		log.Printf("Failed to listen on pair socket: %v", err)
+		return nil, err
+	}
```

Comment:
Close socket in case of error (On my mind trick with `defer` would be useful here)